### DISCUSSION TITLE
Generation 9 data

### DISCUSSION THREAD

**Discussion by wavebeem** [PokeAPI/pokeapi#769] [Ideas]
FYI it looks like Pokémon Database has a bunch of Generation 9 data in a table that would be easy to parse

https://pokemondb.net/pokedex/stats/gen9

I'm planning on updating [pkmn.help](https://pkmn.help) with this data as supplemental to PokéAPI, until Gen 9 data is available here.

**Comment by Naramsim** [comment]
Hi @wavebeem, we currently don't have a team that updates the data. We do it if we have some free time or we rely on contributors. We have some scripts that pull data from Bulbapedia, but non from pokemondb. We will see :)

**Reply by Keeninsky** [reply]
Looks like Bulbapedia has got data too :)

**Reply by Naramsim** [reply]
hmm, then it's a matter of running the script :) Hahah, I'm on vacation so I will have a hard time. We'll see

**Reply by Keeninsky** [reply]
No hurry, the data there isn't complete by now. So enjoy your vacation :)

**Reply by Piaomu** [reply]
Anyone want to team up to scrub the data?

**Reply by Keeninsky** [reply]
I would love to help, at least i could offer german translations :) 
I think by now the bulbapedia data is at a state where we could start of with an export like @Naramsim mentioned above.

**Reply by Piaomu** [reply]
I wrote a script that converted the raw text from Centro Leaks' Pokedex leaks to JSON a few weeks ago and have been using that data for my personal development for the time being, but I'd like a more refined and complete data set moving forward.

I started writing a Serebii Pokedex Web Scraper in C# last weekend only to realize that the license purchase was obscenely expensive (perhaps I'd have better luck with Beautiful Soup and Python?)

But if there's already a script to scrape data from Bulbapedia, it's probably in our best interests to not reinvent the wheel and just update the script to account for Scarlet/Violet changes like Tera types.

I don't wanna step on anyone's toes, but I'd certainly like to help with the process. 

@Naramsim , what's the best way to help support this process?

@Keeninsky awesome sounds good 😊

**Reply by Naramsim** [reply]
Hi, here some scripts: https://github.com/PokeAPI/pokedex/tree/master-pokeapi/pokeapi/scripts

**Reply by Piaomu** [reply]
@Naramsim I studied the scripts over the weekend, cloned the repo, and ran the script -- I have a few questions.

I understand how the script works at a high level--the script runs a pokemon scraper and a species scraper. Those methods pass data from the data folder, replace a variable in the http path in the GET, and scrapes the bulbapedia edit page to populate the data in the out folder.

1. How did you populate the information in the data folder? Did you manually hard-code the names of all of the gen 8 pokemon into the pokemon.txt file and then the script does the rest?
2. I saw pokeapi data is basically populated by updating a bunch of csv files that are either manually entered by users or populated by a script -- how much of that data can be updated by running the golang script, and how much more must be updated by other means?
3. I'd like to get a better understanding of the golang script for the future for a few reasons:
  -- I'm a .NET api developer by trade, and I have a tenuous understanding of Go to begin with, and I'd like to understand the language and script better.
 -- I'd like to contribute to the webscraping project and the pokeapi in meaningful ways (especially since it seems like the webscraping is maintained by very few people).
 -- Once we've scraped gen 9 data and made it publicly available through pokeapi, I'd like to help out with optimization and documentation for the webscraping script -- that way it's something that can be more easily maintained by the community.

All of this to ask -- would you (or anyone else in this tread) be willing to meet for a short time via Zoom or something to walk through the script and explain what data/variables/packages need to be updated each generation to get a clean scrape and what the best practice would be to contribute the scraped data to pokeapi?

Totally understand if that's a big ask--just thought I'd shoot my shot. I'm really excited about the web scraping project and pokeapi.

**Reply by NathanGuidry** [reply]
Is anyone still working on this?

**Reply by Piaomu** [reply]
> Is anyone still working on this?

I reached out with questions about updating the scripts, but I didn't get a response. Would love to update them myself, but I've never worked in Go so I'd rather work collaboratively with someone who understands the language better than myself.

Still would love to help though.

**Reply by yazae** [reply]
Hi @Piaomu , I’m not part of the project but I’m interested in it too and would like to see if I can help it.
Problem is that I have no real competences on python or database languages, we will see x’)

**Reply by Piaomu** [reply]
> Hi @Piaomu , I’m not part of the project but I’m interested in it too and would like to see if I can help it. Problem is that I have no real competences on python or database languages, we will see x’)

The webscraper for gen 8 data was written in Go. I understand how it (mostly) works, but there are a few input/output things I'm a little hung up on. I honestly don't think it'll take much time to update if we had a little guidance on how it works. It's not too deep. Definitely would be willing to get together to work on it though. It'll be nice to have gen 9 data available on a public-facing endpoint.

**Reply by yazae** [reply]
I actually don't understand what is missing yet in the 9G datas, doesn't #783 and #778 handle most of the data ? There is also several pull requests awaiting reviews.

It seems to me that the current need is in someone willing to be part of the PokeAPI organisation, with enough view of the project to verify and, provide support and accept merge requests 🤔 
(I know there is someone active there, but it don't seems to be enough / the person would probably enjoy the help)

**Reply by Naramsim** [reply]
Hi, yesterday I deployed some of the new gen9 data
